add: core profiler fetch extensions

rjchow opened the pull request:
### Submission Review Guidelines:

- I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
- I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
- I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/). 
- Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### Changes proposed in this Pull Request:

<!-- Describe the changes made to this Pull Request and the reason for such changes. -->

Adds a prefetch call in the background for fetching extensions and countries in the init state without blocking the UI state.

This makes it so that subsequently when these calls are made, there won't be any loading time

Closes #37992.

<!-- Begin testing instructions -->

### How to test the changes in this Pull Request:

<!-- Include detailed instructions on how these changes can be tested. Review and follow the guide for how to write high-quality testing instructions. -->

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

Intermediate PR without user facing feature, QA testing not required

1.  Clone repo
2. Go to new profiler page
3. Note that there is no loading time for the 'skip flow'
4. Reload profiler
5. Note that Extensions page has the extensions populated, and also without loading time

<!-- End testing instructions -->

moon0326 reviewed (dismissed):
LGTM and tested well 👍 🚀 

I think we need a rebase.

Pre-approving

rjchow committed:
add: fetch extensions and countries from API for core profiler

- prefetch it once in the machine init so that it's instant when used later
- fetch it again in preExtensions state and store it to context
- added prefetch for getCountries as well

chihsuan reviewed (approved):
Looks good to me! 👍 😃
